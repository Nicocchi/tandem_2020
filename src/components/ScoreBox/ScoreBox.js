import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

function ScoreBox(props) {
    const highScore = localStorage.getItem("score");
    return (
        <div className="score-box">
            <h1>{props.incorrect <= 2 ? "Congrats!" : "Game Over"}</h1>
            <span>
                <p>Correct:</p>
                <p className="score-num" data-testid="scorebox-correct-val">
                    {props.correct}
                </p>
            </span>
            <span>
                <p>Incorrect:</p>
                <p className="score-num" data-testid="scorebox-incorrect-val">
                    {props.incorrect}
                </p>
            </span>
            <span>
                <p>Score:</p>
                <p className="score-num" data-testid="scorebox-score-val">
                    {props.score}
                </p>
            </span>
            <span>
                <p>High Score:</p>
                <p className="score-num" data-testid="scorebox-highscore-val">
                    {highScore ? highScore : props.score}
                </p>
            </span>
            <button className="answer-btn next-btn animated flip-in-y" onClick={props.playAgain}>
                Play Again?
            </button>
        </div>
    );
}

ScoreBox.propTypes = {
    correct: PropTypes.number,
    incorrect: PropTypes.number,
    score: PropTypes.number,
};

const mapStateToProps = (state) => {
    return {
        correct: state.gameReducer.correct,
        incorrect: state.gameReducer.incorrect,
        score: state.gameReducer.score,
    };
};

export default connect(mapStateToProps)(ScoreBox);
