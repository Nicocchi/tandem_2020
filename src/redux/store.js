import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import combineReducer from "./reducers";

let store = null;

// Redux dev tools
const composeEnhancers =
    typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
              // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
          })
        : compose;

const devEnhancer = composeEnhancers(applyMiddleware(thunk, logger));
const enhancer = composeEnhancers(applyMiddleware());

// Create redux store
if (process.env.REACT_APP_NODE_ENV !== "production") {
    store = createStore(combineReducer, devEnhancer);
} else {
    store = createStore(combineReducer, enhancer);
}

export default store;