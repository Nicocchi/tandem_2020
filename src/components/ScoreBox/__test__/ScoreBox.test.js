import React from "react";
import ReactDOM from "react-dom";
import ScoreBox from "./../ScoreBox";
import store from "../../../redux/store";
import { Provider } from "react-redux";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

describe("ScoreBox", () => {
    afterEach(() => {
        cleanup();
    });

    it("renders without crashing without props", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={store}>
                <ScoreBox />
            </Provider>,
            div
        );
    });

    it("matches snapshop", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <ScoreBox />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("question", () => {
        it("renders the text correctly", () => {
            const { getByTestId } = render(
                <Provider store={store}>
                    <ScoreBox />
                </Provider>
            );

            // To be 0 because of initial state is 0
            expect(getByTestId("scorebox-correct-val")).toHaveTextContent("0");
            expect(getByTestId("scorebox-incorrect-val")).toHaveTextContent("0");
            expect(getByTestId("scorebox-score-val")).toHaveTextContent("0");
            expect(getByTestId("scorebox-highscore-val")).toHaveTextContent("0");
        });
    });
});
