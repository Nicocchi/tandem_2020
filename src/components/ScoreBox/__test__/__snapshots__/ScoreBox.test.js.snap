// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBox matches snapshop 1`] = `
<div
  className="score-box"
>
  <h1>
    Congrats!
  </h1>
  <span>
    <p>
      Correct:
    </p>
    <p
      className="score-num"
      data-testid="scorebox-correct-val"
    >
      0
    </p>
  </span>
  <span>
    <p>
      Incorrect:
    </p>
    <p
      className="score-num"
      data-testid="scorebox-incorrect-val"
    >
      0
    </p>
  </span>
  <span>
    <p>
      Score:
    </p>
    <p
      className="score-num"
      data-testid="scorebox-score-val"
    >
      0
    </p>
  </span>
  <span>
    <p>
      High Score:
    </p>
    <p
      className="score-num"
      data-testid="scorebox-highscore-val"
    >
      0
    </p>
  </span>
  <button
    className="answer-btn next-btn animated flip-in-y"
  >
    Play Again?
  </button>
</div>
`;
